src/classes/TriggerAccountHandler.cls

```diff
@@ -0,0 +1,75 @@
+/**
+* Description   - Trigger Account Handler
+* Created by    - Sagar Nirwal
+* Created Date  - 25/07/2022
+* Revision log  -  V_1.0 - Created  - Sagar Nirwal - 25/07/2022
+*                  V_1.1 - Modified - Sagar NIrwal - 25/07/2022 -  Details what you modified
+**/
+public with sharing class TriggerAccountHandler {
+    /**   Description :   Method for Before insert and update  
+    *     Created By  :   Sagar Nirwal
+    *     Arguments   :   List(trigger.new),List(trigger.old)
+    *     Return      :   void
+    *     Revision log:   V_1.0 - Created  - Sagar Nirwal - 25/07/2022
+    *                     V_1.1 - Modified - Sagar Nirwal - 25/07/2022 - Details what you modified
+    **/
+    public static void insertAndUpdateAccountType(List<Account> newList,  List<Account> oldList) {
+        //String variable for removing prevoius SA or FD from account number
+        String prefixLetter;
+        for(Account acc : newList){
+            if(acc.Type != null && acc.Type == 'Saving Account' ){
+                //removing the previous prefix of account number to add the apporpirate prefix after update
+                prefixLetter = (acc.AccountNumber.contains('FD-')) ? acc.AccountNumber.remove('FD-') : acc.AccountNumber ;
+                acc.AccountNumber = 'SA-'+ prefixLetter;
+
+            }
+            if(acc.Type != null && acc.Type == 'Fixed Deposit'){
+                //removing the previous prefix of account number to add the apporpirate prefix after update
+                prefixLetter = (acc.AccountNumber.contains('SA-')) ? acc.AccountNumber.remove('SA-') : acc.AccountNumber ;
+                acc.AccountNumber = 'FD-'+ prefixLetter;
+            }
+            if(acc.Type != null && acc.Type == 'Other'){
+                if(oldList != null && oldList[0].Type == 'Saving Account'){
```

Comment:
bulk data this code not work. Use old map and get the particular record. 
Like OldAccountsMap.get(acc.Id)